src/naip_cnn/data.py

```diff
@@ -513,6 +517,25 @@ def decode_band(band: tf.Tensor) -> tf.Tensor:
         example = tf.io.parse_single_example(serialized_example, features)
         image = tf.stack([decode_band(example[b]) for b in bands], axis=-1)
 
+        # Training data was extracted with NAIP imagery oriented south-up (positive
+        # pixel height). To avoid domain shift during inference, ensure that the
+        # TFRecord imagery is oriented the same way based on the affine transform.
+        # See https://github.com/lemma-osu/naip-cnn/issues/25
+        # Note that GEE affine transform matrices are stored as [a b c d e f] compared
+        # to the GDAL convention of [c a b f d e] or shapely [a b d e c f] where:
+        # a = pixel width
+        # b = row rotation (typically zero)
+        # c = x-coordinate of upper-left pixel
+        # d = column rotation (typically zero)
+        # e = pixel height (typically negative)
+        # f = y-coordinate of upper-left pixel
+        # Also see https://gdal.org/en/stable/tutorials/geotransforms_tut.html
+        if self.profile["transform"][4] < 0:
+            image = tf.image.flip_up_down(image)
+        # This could theoretically happen in the x-direction as well.
+        if self.profile["transform"][0] < 0:
+            image = tf.image.flip_left_right(image)
+

```

Comment:
Again, great to have this information.  For `f`, if `e` is positive, isn't `f` the y-coordinate of the lower-left pixel?

Reply:
Yeah, good point. I changed `c` and `f` to just refer to the "origin pixel", since the location will vary depending on `a` and `e`.